### PR TITLE
Realign filesystem sync documentation for easier learning

### PR DESCRIPTION
**Pull request by grayside**
Tried to push this around into what seems like more of a learning order, with less up-front emphasis on the justification of leaving NFS behind.

Here's the basic outline:

1. What is it?
2. How do you configure docker for it?
3. How does outrigger support it
4. Why aren't we using NFS?

### CODE REVIEW

File: docs/project-setup/filesystem-sync.md [hunk 1]
```diff
@@ -1,41 +1,22 @@
...
-a process in your container to rebuild or update compiled project assets when modification notifications are triggered
-due to local file editing in your IDE.
+From the root of your project, all the files are synced via a dedicated container and exposed as a named [Docker Volume](https://docs.docker.com/engine/tutorials/dockervolumes/) to the rest of your application. This volume is used as the file "source" for a volume mount to allow nearly native filesystem performance and features.
 
-In order to better support these types of operations, Outrigger provides support for multi-directional filesystem syncing 
-(which includes filesystem notifications) using `unison`.  Your project root folder is synced via `unison` into a sync 
-container and exposed as a Docker Volume to the rest of your application. This volume is then used as your mount point 
-instead of the NFS share and allows nearly native filesystem performance and features.
+This is not used as the only means of sharing code or data into the filesystem, but it is the recommended approach for sharing primary application code that must also be readily available to developers and their local tools (e.g., IDEs).
```
Comment (tekante) [inline docs/project-setup/filesystem-sync.md]: I'd reword the first sentence to the following and move this as the end of the first paragraph.

"This is not the only means of sharing code or data with application containers, but it is the recommended . . . "

File: docs/project-setup/filesystem-sync.md [hunk 2]
```diff
@@ -1,41 +1,22 @@
...
-However, NFS can be slow when writing, reading, or scanning thousands of files when compared to native filesystem performance. 
-Some modern tool kits and package managers favor large numbers of small files and libraries so maintaining high performance 
-of a build becomes challenging when your code base is on a volume mounted NFS share.
+Outrigger uses [Unison](https://github.com/bcpierce00/unison) to provide high-performance, bi-directional file synchronization between your local operating system and Docker containers.
 
-NFS also has the downside of not propagating filesystem modification events. This can be problematic if you want to run
-a process in your container to rebuild or update compiled project assets when modification notifications are triggered
-due to local file editing in your IDE.
+From the root of your project, all the files are synced via a dedicated container and exposed as a named [Docker Volume](https://docs.docker.com/engine/tutorials/dockervolumes/) to the rest of your application. This volume is used as the file "source" for a volume mount to allow nearly native filesystem performance and features.
```
Comment (tekante) [inline docs/project-setup/filesystem-sync.md]: Recommend changing the beginning to: "With Unison, all the files from the root of your project are synced via a dedicated container and exposed . . . ."

File: docs/project-setup/filesystem-sync.md [hunk 3]
```diff
@@ -1,41 +1,22 @@
...
-1. Argument provided to the `project sync:start` command
-2. Specified in the `sync` -> `volume-name` property of your [Outrigger Project Configuration](./project-configuration.md)
-3. In your `docker-compose.yml` file, specified as an `external` volume with a name of the pattern `*-sync`
-4. If not specified anywhere else, it will use the name of the current project folder with `-sync` appended
-
-We recommend using approach #3, as it is the most explicit and is in line with all of the configuration you will need
-to take full advantage of `unison` syncing.
+To setup `unison` sync for your containers you will need to [reference an external volume](https://docs.docker.com/compose/compose-file/#volume-configuration-reference)
+in your `docker-compose.yml` and use that external volume in the mount specification for any services that need to
+reference the same set of files. Using the same volume will ensure all containers stay in sync. This includes specifying that volume for your [Build container](/common-tasks/using-the-build-container), if you are using that (via build.yml).
```
Comment (tekante) [inline docs/project-setup/filesystem-sync.md]: Before the "Using the same volume " part it would be helpful to add some naming information. Suggestion:

Since volumes are global within your Docker host, the name of this external volume should be name-spaced to your project. The convention is to use projectname-sync as the volume name. Using the same volume will ensure all of your application's containers stay in sync. This includes ....

File: docs/project-setup/filesystem-sync.md [hunk 4]
```diff
@@ -44,13 +25,13 @@ volumes:
 ```
 
 !!! important "Note about volume name"
-    If you optionally specified a volume name to the `project sync:start` command or configured a name in your Outrigger 
+    If you optionally specified a volume name to the `project sync:start` command or configured a name in your Outrigger
```
Comment (tekante) [inline docs/project-setup/filesystem-sync.md]: In the example above, naming the volume "projectname-sync:" will help make it more clear that one shouldn't use the literal value "project-sync" there.

File: docs/project-setup/filesystem-sync.md [hunk 5]
```diff
@@ -60,19 +41,79 @@ services:
       - project-sync:/var/www
 ```
 
-## Start it up
+## Managing the File Sync Process
```
Comment (tekante) [inline docs/project-setup/filesystem-sync.md]: Same projectname-sync recommendation here.

File: docs/project-setup/filesystem-sync.md [hunk 6]
```diff
@@ -60,19 +41,79 @@ services:
...
 When you are done for the day, or for that project, from the project root `rig project sync:stop` will clean up any running
 `unison` containers for that project.
+
+## Skipping File Sync on Linux Environments
+
+Linux environments have native filesystem performance without anything as complex as file sync, because they do not need to use a virtualization layer to have access to the [Linux Kernel's container support](https://jvns.ca/blog/2016/10/10/what-even-is-a-container/).
+
+For this reason, any environment running on Linux may want to skip the overhead of the synchronization process.
+
+If you have this use case, create a  docker-compose.override.yml file to carry the external volume declaration, as described in the [official documentation for sharing docker-compose configuration](https://docs.docker.com/compose/extends/#multiple-compose-files). Unlike that documentation, on Linux environments you simply run as `docker-compose -f docker-compose.yml` and the configuration override will not be applied.
```
Comment (febbraro) [inline docs/project-setup/filesystem-sync.md]: I think you need a new header here? Or is this still about skipping linux? ;)

File: docs/project-setup/filesystem-sync.md [hunk 7]
```diff
@@ -60,19 +41,79 @@ services:
...
+      - .:/var/www
+```
+
+### Example docker-compose.override.yml
+
+```yaml
+services:
+  www:
+    volumes:
+      - project-sync:/var/www
```
Comment (febbraro) [inline docs/project-setup/filesystem-sync.md]: projectname-sync

File: docs/project-setup/filesystem-sync.md [hunk 8]
```diff
@@ -60,19 +41,79 @@ services:
...
+volumes:
+  project-sync:
```
Comment (febbraro) [inline docs/project-setup/filesystem-sync.md]: projectname-sync

**Review by febbraro — APPROVED** [review]
A few tweaks but otherwise good!

File: docs/project-setup/filesystem-sync.md (hunk 6 above)
Comment (grayside) [reply docs/project-setup/filesystem-sync.md]: Yes, this is the how-to on how to set this up to support Linux users instead of just assuming always use filesync.

**Comment by grayside** [comment]
Updated to incorporate feedback, and ran the spellchecker. I decided to keep IDEs and namespaced as-is.

**Review by mike-potter — APPROVED** [review]

**Review by crittermike — APPROVED** [review]

**Review by tekante — APPROVED** [review]
I added some tweaks to reduce the back and forth cycle so you may want to check my latest commit before merging but overall looks good to me now.
